### PR TITLE
MOB-4758 read json from GitHub

### PR DESCRIPTION
**Pull request by BartlomiejSopalaKarhoo**
## JIRA
[MOB-4758](https://karhoo.atlassian.net/browse/MOB-4758)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

[MOB-4758]: https://karhoo.atlassian.net/browse/MOB-4758?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+//  KarhooUISDK
+//
+//  Created by Bartlomiej Sopala on 15/04/2023.
+//  Copyright © 2023 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class FeatureFlagsUpdater {
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift:12]: In the sake of consistency, I'd prefer using `Service`

File: KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        self.featureFlagsStore = featureFlagsStore
+    }
+    
+    func update() {
+        let jsonUrl = "https://raw.githubusercontent.com/karhoo/karhoo-ios-ui-sdk/MOB-4757-feature-flag-file/KarhooUISDK/FeatureFlags/feature_flag.json"
+
+        let url = URL(string: jsonUrl)!
+        let decoder = JSONDecoder()
+        
+        let task = URLSession.shared.dataTask(with: url) {[weak self] data, _, error in
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift:31]: whitespace required between `{` and `[`

File: KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    
+    init(
+        currentSdkVersion: String,
+        featureFlagsStore: FeatureFlagsStore = KarhooFeatureFlagsStore()
+    ) {
+        self.currentSdkVersion = currentSdkVersion
+        self.featureFlagsStore = featureFlagsStore
+    }
+    
+    func update() {
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift:25]: What are your thoughts on ensuring this whole update flow is performed on utility QoC?

File: KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                }
+            } else if let error {
+                // TODO: Add error to analytics
+            }
+        }
+        task.resume()
+    }
+    
+    func handleFlagSets(_ sets: [FeatureFlagsModel]) {
+        if let selectedSet = selectProperSet(forVersion: currentSdkVersion, from: sets) {
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift:47]: main method flow -> `guard` preferred

File: KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            storeFeatureFlag(selectedSet)
+        }
+    }
+    
+    private func selectProperSet(forVersion version: String, from sets: [FeatureFlagsModel]) -> FeatureFlagsModel? {
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift:52]: Afaik we should iterate through all sets and get the newest data for every property, shouldn't we?

Comment (BartlomiejSopalaKarhoo) [reply KarhooUISDK/FeatureFlags/FeatureFlagsUpdater.swift:52]: I don't think so. Let's talk about it tomorrow.

**Review by dianakarh — APPROVED** [review]

**Pull request merged**